### PR TITLE
Centrys 49

### PR DESCRIPTION
**Pull request by alinturbut**

### CODE REVIEW

File: tests/paymentProcessor/testPaymentProcessorTwoMiners.js [hunk 1]
```diff
@@ -50,8 +53,8 @@ setTimeout(verifyBothMinersGetPaidFromTimeToTime, 180000);
 
 let cleanupFunction = () => {
     console.log('cleaning up the processes');
-    subprocess1.kill();
-    subprocess1.kill();
+    minerProcess1.kill();
+    minerProcess1.kill();
```
Comment (alexandrulaurus) [inline tests/paymentProcessor/testPaymentProcessorTwoMiners.js]: Shouldn't it be `minerProcess2` here?

File: tests/miningPool/testMiningPoolMultiplePools.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    //start first miner
+    let argument1 = ["-t", testConfig.miner1Threads, "-u", testConfig.miner1];
+    minerProcess1 = spawn(testConfig.aionminerCpuLocation, argument1, { detached: true});
+
+    minerProcess1.on('error', function(err){
+        console.log(err);
+    });
+
+    //start second miner
+    let argument2 = "-t " + testConfig.miner2Threads + " -u " + testConfig.miner2;
```
Comment (alexandrulaurus) [inline tests/miningPool/testMiningPoolMultiplePools.js]: Can you be consistent with the argument passing as `argument1`

File: tests/miningPool/testMiningPoolMultiplePools.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        console.log(err);
+    });
+};
+
+let startMiners = () => {
+    //start first miner
+    let argument1 = ["-t", testConfig.miner1Threads, "-u", testConfig.miner1];
+    minerProcess1 = spawn(testConfig.aionminerCpuLocation, argument1, { detached: true});
+
+    minerProcess1.on('error', function(err){
```
Comment (alexandrulaurus) [inline tests/miningPool/testMiningPoolMultiplePools.js:28]: I find it difficult to find what's wrong just by logging errors. What do you think about redirecting `stderr` and `stdout` and seeing the actual logs for each subprocess(miner? Something like:
```
minerProcess1.stderr.on('data', function(data) {
    //you may need to check for multiple line endings here
    console.log(data);
});
```

File: tests/paymentProcessor/testPaymentProcessorTwoMiners.js [hunk 4]
```diff
@@ -3,26 +3,26 @@ let web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
...
+let minerProcess2;
 
 let startBothMiners = () => {
     //start first miner
     let argument1 = ["-t", testConfig.miner1Threads, "-u", testConfig.miner1];
-    subprocess1 = spawn(testConfig.aionminerCpuLocation, argument1, { detached: true});
+    minerProcess1 = spawn(testConfig.aionminerCpuLocation, argument1, { detached: true});
 
-    subprocess1.on('error', function(err){
+    minerProcess1.on('error', function(err){
```
Comment (alexandrulaurus) [inline tests/paymentProcessor/testPaymentProcessorTwoMiners.js:17]: Same comment regarding output redirection

**Review by alexandrulaurus — CHANGES_REQUESTED** [review]

File: tests/miningPool/testMiningPoolMultiplePools.js (hunk 3 above)
Comment (alinturbut) [reply tests/miningPool/testMiningPoolMultiplePools.js:28]: node & spawn will complain if we don't catch the error like this from what I remember

**Review by centrys-george — APPROVED** [review]

File: tests/miningPool/testMiningPoolMultiplePools.js (hunk 3 above)
Comment (alexandrulaurus) [reply tests/miningPool/testMiningPoolMultiplePools.js:28]: You could leave the error handling and also add the output listeners afterwards.

**Review by alexandrulaurus — APPROVED** [review]

**Pull request merged**